### ISSUE TITLE
Fiberassign "survey log"

### ISSUE THREAD

**Issue by tskisner** [desihub/fiberassign#276]
In order to "replay" fiber assignment on simulated data, we need a kind of "survey log" that has for each time the assignment was run:
- The run date, which will be used to get the focalplane state from desimodel and the target state from the MTL ledger.
- The list of tile IDs that were assigned (regardless of which ones are actually observed later) 
I believe that all this info can be obtained from the headers of the output fiberassign files stored someplace like the tiles svn repository.  We just need to have a script that scans over the headers and makes a list of tiles for each run date.

Question:  where should this script live?  It could be in a repo related to survey planning or one that is related to simulations since it is external to the survey operations.

One benefit to having it in the fiberassign repo is that it would be easier to update if header keys change, etc.  Thoughts?

**Comment by ashleyjross** [comment]
This is something I can probably take on, but the question (and some related ones) probably needs some discussion. The problem is exactly what we decide to do with operations affects what we need to do here. (For example, if we go with "no pass" or not.)

A related issue is what the plan is for exactly where the fiberassign files get stored/organized. These will be on svn repo, with a different directory for each fiberassign run, I assume, and organized by date?

**Comment by tskisner** [comment]
Currently the tiles repo:

https://desi.lbl.gov/trac/browser/data/tiles/trunk

Is a bucket of tiles sorted into sub directories based on the first 3 digits of tile ID.  I think we just need to load these keys:

https://github.com/desihub/fiberassign/blob/master/py/fiberassign/assign.py#L257-L263

from each output file, and then group those tile properties by the value of `FA_RUN` key.  This way, we can actually construct a fiberassign `Tiles` object directly from that info.

Not sure the best output format.  In memory it would look like a dictionary with FA_RUN values as keys and the values would be a list of tiles with those other properties.  Heck it won't be that big so dumping to JSON or YAML might be fine.  For FITS, I guess one could have a different HDU for each FA_RUN value, but that would end up being a lot of HDUs...

**Comment by ashleyjross** [comment]
I guess this is my point: That current organization clearly will not be convenient for what we want to do? The same tile will be fiberassigned on different days and we have to find the correct one. (Maybe no pass fixes this, since the last time something got fiberassigned would be what was observed in that case and also what was in trunk, but I would rather have a more explicit connection.)

**Comment by tskisner** [comment]
I agree that if we are going to assign the same tile ID multiple times, either the files with the same ID must be in different directories or have some other string in the name.  Another option would be to "retire" tile IDs once they are assigned (not just after they are observed) and every assignment run would use new tile IDs for a given pointing location on the sky.

**Comment by michaelJwilson** [comment]
I'd have thought we'd associate this log to the mtl itself.  E.g. as a dedicated hdu in the initial ledger file.  I would have thought updating this with run date, tile list and output filepaths would be done on each call by fiberassign itself.

There's the usual problem of how the sims should work round this, e.g. they won't be committing to svn.  My two cents.
